### DISCUSSION TITLE
Questions and suggestions about the UI

### DISCUSSION THREAD

**Discussion by ovizii** [karakeep-app/karakeep#76] [Ideas]
I'm wondering why clicking on the preview (1) of a bookmark sends me straight to the "Original" while wanting to read the bookmark makes me look for that "small" expand icon (2). 

Isn't the default action to read the crawled content, thus I intuitively always click the preview which seems to defeat the whole purpose of crawling, storing and read-later.

![image](https://github.com/MohamedBassem/hoarder-app/assets/837902/840d8917-c195-44ec-a8f3-f774189941e8)

Also, wouldn't it make sense to display all those actions shown in (3) also in view (1) of the second image?

![image](https://github.com/MohamedBassem/hoarder-app/assets/837902/c0e0f132-bc64-4cf3-8432-f295e3d5d9ec)

**Comment by MohamedBassem** [comment]
> I'm wondering why clicking on the preview (1) of a bookmark sends me straight to the "Original" 

I thought about this a bit while designing it. In my opinion, there are two separate usecases:
1. If you're treating Hoarder as a bookmarking app (and some of the friends on my selfhosted app are), you'd probably want the current behavior. Clicking on the link goes to the original.
2. If you're treating Hoarder as a read-it-later app, you probably want to read the crawled content.

To be honest, rendering the crawled content doesn't always look great especially for non-article-like kind of links, which is what the `bookmarking` usecases typically save. That's why I settled on going to the original as a safe default. Given that hoarder fits both in the `bookmarking` and the `read-it-later` category, I think this behavior can potentially be configurable per user.

> which seems to defeat the whole purpose of crawling, storing and read-later.

Kinda defeats the purpose for read-it-later I agree. But for crawling and storing, the other important purpose of that beside read-later is indexing for full text search.

I think once the `preview` page looks good for all type of content, I might reconsider this decision. I don't have strong opinions on that anyways.

> Also, wouldn't it make sense to display all those actions shown in (3) also in view (1) of the second image?

This is on my todo list and it'll probably happen in the next release!
Also, btw, you might want to upgrade to `v0.11.0`, there's now a `Notes` field where you drew your rectangle ;)

**Reply by MohamedBassem** [reply]
Oh, and thanks a lot for taking the time and sharing your suggestions. Please keep them coming!

**Comment by ovizii** [comment]
> I thought about this a bit while designing it. In my opinion, there are two separate usecases:
> 
> If you're treating Hoarder as a bookmarking app (and some of the friends on my selfhosted app are), you'd probably want the current behavior. Clicking on the link goes to the original.
> If you're treating Hoarder as a read-it-later app, you probably want to read the crawled content.

I agree, you're right.

> Kinda defeats the purpose for read-it-later I agree. But for crawling and storing, the other important purpose of that beside read-later is indexing for full text search.
> 
> I think once the preview page looks good for all type of content, I might reconsider this decision. I don't have strong opinions on that anyways.

Great, I'm looking forward to it.

> This is on my todo list and it'll probably happen in the next release!
> Also, btw, you might want to upgrade to v0.11.0, there's now a Notes field where you drew your rectangle ;)

I noticed there's no version info in the backend, would be nice to see it in the admin section maybe? Possibly with an indicator about newer versions being available and a link to the release?

**Reply by MohamedBassem** [reply]
The version info and the new release check are both included in v0.11.0 in exactly the way your described them ;)
